presentation/Main.java

```diff
@@ -0,0 +1,69 @@
+package presentation;
+
+import dao.Dao;
+import dao.StudentDao;
+import model.Student;
+import service.StudentService;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Main {
+
+    public static void main(String[] args) {
+        int userOption;
+        Scanner scanner = new Scanner(System.in);
+        Dao<Student> dao = new StudentDao();
+        StudentService studentService = new StudentService(dao);
+        Menu menu = new Menu();
+        do{
+            menu.showMenu();
+            userOption= scanner.nextInt();
+            switch (userOption) {
+                case Menu.ADD_STUDENT -> {
+                    Student studentToAdd = new Student();
+                    System.out.println("Enter the Student Details: ");
+                    System.out.print("Student Id: ");
+                    studentToAdd.setStudentId(scanner.next());
+                    System.out.print("First Name: ");
+                    studentToAdd.setFirstName(scanner.next());
+                    System.out.print("Last Name: ");
+                    studentToAdd.setLastName(scanner.next());
+                    System.out.println();
+                    studentService.addStudent(studentToAdd);
+                    System.out.println();
+                }
+                case Menu.UPDATE_STUDENT -> {
+                    System.out.print("Enter the StudentId to update: ");
+                    String studentId = scanner.next();
+                    Student student = studentService.getStudentById(studentId);
+                    Student student1 = new Student();
```

Comment:
`student1` is not really a meaningful name. When I read this code, `student` represents `oldStudent` or `oldStudentValues`. `student1` represents `newStudent` or `newStudentValues` in this case. Also, you do not really need 2 variables to store the old values and new values. You can also get the student from the service by calling `studentService.getStudentById(studentId)` and set the new values for the `firstName` and `lastName` by using the setters.
The flow is anyways **print student old last name** -> **replace student old last name with user input**. After that you do not need the old value of the student's last name anymore.

```java
Student student = studentService.getStudentById(studentId);
System.out.printf("Enter the First Name (%s): ", student.getFirstName()); // student.firstName = "Test"
student.setFirstName(scanner.next()); // student.firstName = "Raimund"
```

If you need the old student's `firstName` value afterward, then your implementation makes total sense. Just do not forget to apply meaningful naming.

Reply:
`Student student = studentService.getStudentById(studentId);`
`student.setFirstName(scanner.next());`
Here I'm getting the reference of student object (by id), If i change the values using the setters, then what is the use of `void update(T newValues);` parameters in update method.

Reply:
That is a really good question.
First of all, we need the update method since your `Student get(String id)` method in `StudentDao` returns a new object instead of a reference. And second imagine that instead of an `ArrayList`, we have a database and a `MySqlStudentDao` for example that writes the new values to the database by using SQL statements (`UPDATE Student SET firstName=?, lastName=? WHERE studentId=?;`).

![image](https://user-images.githubusercontent.com/12022545/223210686-23613320-5fb0-41b3-8951-ce523f1cf216.png)